base/loggers/OwnPatternFormatter.cpp

```diff
@@ -7,13 +7,118 @@
+    writeCString("\"", wb);
+    writeCString(":", wb);
+    writeCString("\"", wb);
+    /// Change delimiters in date for compatibility with old logs.
+    DB::writeDateTimeText<'.', ':'>(msg_ext.time_seconds, wb);
+
+    DB::writeChar('.', wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 100000) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 10000) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 1000) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 100) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 10) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 1) % 10), wb);
+    writeCString("\"", wb);
+
+    writeCString(",", wb);
+
+    writeCString("\"", wb);
+    writeCString("thread_id", wb);
+    writeCString("\"", wb);
+    writeCString(":", wb);
+    writeCString("\"", wb);
+    writeCString(" [ ", wb);
+    if (color)
+        writeString(setColor(intHash64(msg_ext.thread_id)), wb);
+    DB::writeIntText(msg_ext.thread_id, wb);
+    if (color)
+        writeCString(resetColor(), wb);
+    writeCString(" ] ", wb);
+    writeCString("\"", wb);
+
+    /// We write query_id even in case when it is empty (no query context)
+    /// just to be convenient for various log parsers.
+    writeCString(",", wb);
+
+    writeCString("\"", wb);
+    writeCString("query_id", wb);
+    writeCString("\"", wb);
+    writeCString(":", wb);
+    writeCString("\"", wb);
+    writeCString("{", wb);
+
+    if (color)
+        writeString(setColor(std::hash<std::string>()(msg_ext.query_id)), wb);
+    DB::writeString(msg_ext.query_id, wb);
+    if (color)
+        writeCString(resetColor(), wb);
+    writeCString("}", wb);
+    writeCString("\"", wb);
+
+    writeCString(",", wb);
+
+    writeCString("\"", wb);
+    writeCString("priority", wb);
+    writeCString("\"", wb);
+    writeCString(":", wb);
+
+    writeCString("\"", wb);
+    writeCString("<", wb);
+
+    int priority = static_cast<int>(msg.getPriority());
+    if (color)
+        writeCString(setColorForLogPriority(priority), wb);
+    DB::writeString(getPriorityName(priority), wb);
+    if (color)
+        writeCString(resetColor(), wb);
+    writeCString(">", wb);
+    writeCString("\"", wb);
+
+    writeCString(",", wb);
+
+    writeCString("\"", wb);
+    writeCString("source", wb);
+    writeCString("\"", wb);
+    writeCString(":", wb);
+
+    writeCString("\"", wb);
+
+    if (color)
+        writeString(setColor(std::hash<std::string>()(msg.getSource())), wb);
+    DB::writeString(msg.getSource(), wb);
+    if (color)
+        writeCString(resetColor(), wb);
+    writeCString("\"", wb);
+
+    writeCString(",", wb);
+
+    writeCString("\"", wb);
+    writeCString("message", wb);
+    writeCString("\"", wb);
+    writeCString(":", wb);
+
+    writeCString("\"", wb);
+    DB::writeString(msg.getText(), wb);
+    writeCString("\"", wb);
+
+    writeCString("}", wb);
```

Comment:
Missing file, line and other fields from system.text_log.

Reply:
Agreed. Will add them. Is there any suggestion of their sequence?

Reply:
The ordering can be identical to the `system.text_log` table.

Reply:
change submitted in updated PR.